dace/libraries/blas/utility/reductions.py

```diff
@@ -0,0 +1,845 @@
+
+    store_task = store_state.add_tasklet(
+        'storeRed_task',
+        ['inCon'],
+        ['outCon'],
+        'outCon = inCon'
+    )
+
+    store_state.add_memlet_path(
+        res, store_task,
+        dst_conn='inCon',
+        memlet=Memlet.simple(res.data, '0')
+    )
+
+    store_state.add_memlet_path(
+        store_task, out,
+        src_conn='outCon',
+        memlet=Memlet.simple(
+            out.data,
+            "ii",
+            wcr_str="lambda a, b: a + b"
+        )
+    )
+
+    # ---------- ----------
+    # Stream out
+    # ---------- ----------
+    y_in = write_y_state.add_read('_y_red')
+
+    write_y_state.add_memlet_path(
+        y_in, y_out_stream,
+        memlet=Memlet.simple(y_in.data, "ii", other_subset_str='0')
+    )
+
+
+    redTile_sdfg.add_edge(store_state, write_y_state, InterstateEdge("j == ({0}/{1}) - 1".format(m, mTile)))
+    redTile_sdfg.add_edge(write_y_state, end_state, InterstateEdge(None))
+    redTile_sdfg.add_edge(store_state, write_empty_state, InterstateEdge("j != ({0}/{1}) - 1".format(m, mTile)))
+    redTile_sdfg.add_edge(write_empty_state, end_state, InterstateEdge(None))
+
+    redTile_sdfg.add_edge(compute_state, red_state, InterstateEdge(None))
+    redTile_sdfg.add_edge(red_state, store_state, InterstateEdge(None))
+
+    redTile_sdfg.fill_scope_connectors()
+
+    return redTile_sdfg
+
+
+
+
+def make_unrolledCompute(dtype, nTile, mTile, veclen, partialWidth, n, m, a):
+
+    inner_sdfg = SDFG("partial_reduction_inner")
+
+    inner_sdfg.add_symbol(a.name, a.dtype)
+
+    init_state = inner_sdfg.add_state("init_state")
+    compute_state = inner_sdfg.add_state('compute_state')
+    write_state = inner_sdfg.add_state("write_state")
+
+    read_x_state = inner_sdfg.add_state("readX_state")
+    read_empty_state = inner_sdfg.add_state("readEmpty_state")
+
+    vec_type = vector(dtype, veclen)
+    singleton_vec = vector(dtype, 1)
+    A_in = init_state.add_stream(
+        '_A_unroll',
+        vec_type,
+        buffer_size=32,
+        storage=dtypes.StorageType.FPGA_Local
+    )
+
+    x_in = read_x_state.add_stream(
+        '_x_unroll',
+        vec_type,
+        buffer_size=32,
+        storage=dtypes.StorageType.FPGA_Local
+    )
+
+    inner_sdfg.add_array('_buf_in_unroll', shape=[max(2, partialWidth)], dtype=dtype,
+        storage=dtypes.StorageType.FPGA_Local if partialWidth > 8 else dtypes.StorageType.FPGA_Registers
+    )
+    inner_sdfg.add_array('buf_out', shape=[max(2, partialWidth)], dtype=dtype,
+        storage=dtypes.StorageType.FPGA_Local if partialWidth > 8 else dtypes.StorageType.FPGA_Registers
+    )
+
+    inner_sdfg.add_array('_x_buf', shape=[mTile], dtype=dtype, storage=dtypes.StorageType.FPGA_Local)
+    inner_sdfg.add_array('vecBuf_x', shape=[veclen], dtype=dtype, storage=dtypes.StorageType.FPGA_Registers, transient=True)
+    inner_sdfg.add_array('memBuf_x', shape=[veclen], dtype=dtype, storage=dtypes.StorageType.FPGA_Registers, transient=True)
```

Comment:
Please run `yapf` (version 0.30.0 or higher) on all the code files to format them

Reply:
Fixed